libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -133,8 +133,10 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
      * @return true if the raw location was usable, false if not.
      */
     override suspend fun updateLocation(rawLocation: FixLocation): Boolean =
-        withContext(NavigatorDispatcher) {
-            navigator!!.updateLocation(rawLocation)
+        suspendCoroutine { continuation ->
```

Comment:
I recommend using `suspendCancellableCoroutine`, because with `suspendCoroutine` you won't be able to cancel the job. 
For example in this case the job still requires 3 seconds to complete and the log is always printed:
```kotlin
withTimeoutOrNull(1000) {
    suspendCoroutine<Unit> { continuation ->
        thread {
            Thread.sleep(3000)
            continuation.resume(Unit)
            println("resumed")
        }
    }
}
```
But if you replace suspendCoroutine with suspendCancellableCoroutine, it will finish after a second and the log won't be printed.